packages/x-charts/src/internals/getWordsByLines.ts

```diff
@@ -27,9 +31,14 @@ export interface GetWordsByLinesParams {
   needsComputation?: boolean;
 }
 
-export function getWordsByLines({ style, needsComputation, text }: GetWordsByLinesParams) {
+export function getWordsByLines({
+  className,
+  style,
+  needsComputation,
+  text,
+}: GetWordsByLinesParams) {
   return text.split('\n').map((subText) => ({
     text: subText,
-    ...(needsComputation ? getStringSize(subText, style) : { width: 0, height: 0 }),
+    ...(needsComputation ? getStringSize(subText, { className, style }) : { width: 0, height: 0 }),
```

Comment:
We were getting an incorrect size because not all styles that are applied to the tick label were being applied to the measurement span. By adding the class, the measurement is now correct.